libs/utils/analysis/frequency_analysis.py

```diff
@@ -218,6 +218,107 @@ def plotClusterFrequencies(self, title='Clusters Frequencies'):
 
         return (avg_lfreq/1e3, avg_bfreq/1e3)
 
+    def plotCpusFrequencies(self, cpus=None):
+        """
+        Plot frequency trend for the specified CPUs (or all is not specified).
+        If sched_overutilized events ar available, the plots will also show the
+        intervals of time where the CPU was overutilized.
+
+        :param cpus: the list of CPUs to plot
+        :type cpus: int or list(int)
+        """
+        if not self._trace.hasEvents('cpu_frequency'):
+            self._log.warning('Events [cpu_frequency] not found, plot DISABLED!')
+            return
+        df = self._dfg_trace_event('cpu_frequency')
+
+        pd.options.mode.chained_assignment = None
```

Comment:
Can't we get rid of the warnings by fixing the code? If not, we should save the old value of `chained_assignment` rather than setting it back to a hard-coded value.